### PR TITLE
Update oval-definitions-schema.xsd to support encapsulated definitions along with 5.8 style defintitions

### PR DESCRIPTION
**Pull request by vanderpol**
Committing the addition of the new encapsulated_definition and  EncapsulatedDefinitionType to preserve work and allow review.

Remaining to be addressed.
1 - In the EncapsulatedDefinitionType, the 'xsd:unique name="UniqueAffectedFamily"' is commented out.  Need to figure out best way to enforce this requirement. 2 - Review that modified 'testKeyRef' checks are appropriate.

### CODE REVIEW

**Comment by vanderpol** [comment]
@johnulmer-oval would like some feedback on his current attempts to support this.  Please review his commit notes/comments/questions.  There is some minor issue with metadata/UniqueAffectedFamily that he would particularly like some feedback on.  

Note that the branch used for development of this was created before we switched to the URN style namespaces, which may be easier for your debugging as well, less change.

Hoping everyone isn't already on vacation for the holidays.  This is our last 'big' item to be resolved before we finalize OVAL 6.0

**Comment by vanderpol** [comment]
Sorry to beg for feedback, but this is pretty critical to OVAL 6.0, if you have a few minutes, we'd really like your input @solind, @A-Biggs, @maxullman @balleman-ctr

File: oval-schemas/oval-definitions-schema.xsd [hunk 1]
```diff
@@ -239,6 +240,73 @@
...
+        <xsd:sequence>
+            <xsd:element ref="ds:Signature" minOccurs="0" maxOccurs="1"/>
+            <xsd:element name="metadata" type="oval-def:MetadataType">
+                <!-- xsd:unique name="UniqueAffectedFamily">
+                    <xsd:annotation>
+                        <xsd:documentation>Each affected element must have a unique family attribute value.</xsd:documentation>
+                    </xsd:annotation>
+                    <xsd:selector xpath="oval-def:affected"/>
+                    <xsd:field xpath="@family"/>
+                </xsd:unique -->
```
Comment (maxullman) [inline oval-schemas/oval-definitions-schema.xsd:276]: What's the issue here for why this is commented out?  Unless I'm missing something, it seems no better or worse than the current validation for the old definition type.

Comment (vanderpol) [reply oval-schemas/oval-definitions-schema.xsd:276]: @johnulmer-oval posted his answer on it back on the issue #181 (he's new to dealing with PR's and Github),  I'll re-post his comments here for easier reading.   Simple answer is that we just need to change them from being unique to elements, as they can no longer be validated as unique.

From @johnulmer-oval :
Have modified oval-definitions-schema.xsd in the '181-update-structure-of-oval-definitions-to-support-encapsulated-defintions...' branch to support encapsulated_definition(s).

Remaining to be addressed.
1 - In the EncapsulatedDefinitionType, the 'xsd:unique name="UniqueAffectedFamily"' is commented out. Need to figure out best way to enforce this requirement.
2 - Review that the modified 'testKeyRef' checks are appropriate.

An example of content employing an encapsulate_definition is attached.

Review and comment is requested.
[simplest_oval6_test.txt](https://github.com/user-attachments/files/18131615/simplest_oval6_test.txt)

As OVAL processing is unchanged from an OVAL 5.11 'definition' to an 'encapsulated_definition', OVAL results are typical.

@[johnulmer-oval](https://github.com/johnulmer-oval)
Member
johnulmer-oval commented [4 hours ago](https://github.com/OVAL-Community/OVAL/issues/181#issuecomment-2549592737)
RE: the 1st 'remaining' issue mentioned in my previous - the 'xsd:unique name="UniqueAffectedFamily"' in the EncapsulatedDefinitionType

When making the EncapsualteDefinitionType, I used the DefinitionType as a template. That brought along the metadata chunk and with it the xsd:unique for the UniqueAffectedFamily. This 'xsd:unique' cannot exist twice with the same element name. So, I'm trying to understand the intended use of the 'family'. After surfing as much documentation as I can find and some content streams, it appears to be mainly a reporting label. If this is the case, I think we can change the xsd:unique to a simple element and the non-uniqueness is eliminated. But, 'would appreciate any input others might have regarding the actual in practice use of the 'family'.

**Comment by johnulmer-oval** [comment]
Re: the xsd:unique check on the family in the metadata, I moved the twin conflicting 'xsd:unique' checks from the 'definition' and 'encapsulated_definition' areas down into the 'metadataType'.

File: oval-schemas/oval-definitions-schema.xsd (hunk 1 above)
Comment (maxullman) [reply oval-schemas/oval-definitions-schema.xsd:276]: For all the content we've seen all the definitions in a given file have the same family.  I can only recall one time seeing a file with both family=unix definitions and some family=windows definitions in the same file, and even in that file each definition only had one family.  It sounds like @johnulmer-oval found a different solution but worst case I don't think it's exceptionally important the schema enforce anything for this attribute.

**Review by maxullman — APPROVED** [review]

**Pull request merged**